code/modules/client/preferences.dm

```diff
@@ -105,7 +105,16 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 		load_path(parent.ckey)
 		if(load_and_save && !fexists(path))
 			try_savefile_type_migration()
-		unlock_content = !!parent.IsByondMember()
+		var/byond_member = parent.IsByondMember()
+		if(isnull(byond_member)) // Connection failure, retry once
+			byond_member = parent.IsByondMember()
+			var/static/admins_warned = FALSE
+			if(!admins_warned)
+				admins_warned = TRUE
+				message_admins("Byond membership lookup had a connection failure for a user. This is most likely an issue on the byond side but if this consistently happens you should bother your server operator to look into it.")
+			if(isnull(byond_member)) // Retrying didn't work, warn the user
+				to_chat(parent, span_warning("There's been a connection failure while trying to check the status of your byond membership. Reconnecting may fix the issue, or byond could be experiencing downtime."))
```

Comment:
Maybe byond should probably be used as a proper noun (Byond) or as its stylized version (BYOND)? obviously not important but it irks me for some reason.

Reply:
Should probably be BYOND in any user-facing text.